clicache/integration-test2/GarbageCollectCache.cs

```diff
@@ -0,0 +1,177 @@
+                    .region()
+                    .withName("testRegion")
+                    .withType("PARTITION")
+                    .execute());
+
+                const int numEntries = 1000;
+                const int numIterations = 100;
+                const int stringLength = 1000;
+
+
+                Int64[] workingSetBefore = new Int64[numIterations];
+                Int64[] workingSetAfter = new Int64[numIterations];
+                string value = new string('*', stringLength);
+
+                for (int i = 0; i < numIterations; i++)
+                {
+                    workingSetBefore[i] = (Int64)Process.GetCurrentProcess().WorkingSet64;
+
+                    using (var cache = new CacheFactory()
+                        .Set("log-level", "none")
+                        .Create())
+                    {
+
+                        cluster.ApplyLocators(cache.GetPoolFactory()).Create("default");
+
+                        var regionFactory = cache.CreateRegionFactory(RegionShortcut.PROXY)
+                            .SetPoolName("default");
+
+                        var region = regionFactory.Create<object, object>("testRegion");
+
+                        IDictionary<object, object> collection = new Dictionary<object, object>();
+
+                        for (int n = 0; n < numEntries; n++)
+                        {
+                            collection.Add(n, value);
+                        }
+
+                        region.PutAll(collection);
+
+                        List<Object> keys = new List<Object>();
+                        for (int item = 0; item < numEntries; item++)
+                        {
+                            Object K = item;
+                            keys.Add(K);
+                        }
+                        Dictionary<Object, Object> values = new Dictionary<Object, Object>();
+                        region.GetAll(keys.ToArray(), values, null, true);
+
+                        cache.Close();
+                    }
+
+                    // Because the are many small leaks in the native client, the
+                    // heuristic below is based on empirical data on across many iterations
+                    // to ensure there are no huge mmemory leaks. After a warmup,
+                    // measurements are taken across several iterations
+                    // to see the trend across many garbage collect cycles.
+                
+                    //if (i > 10)
+                    //    Assert.True(.9 < ratio && ratio < 1.1);
+
+                    workingSetAfter[i] = (Int64)Process.GetCurrentProcess().WorkingSet64;
+
+                }
+
+                Int64[] diff = new Int64[numIterations];
+                for (int i=0; i<numIterations; i++)
+                {
+                    diff[i] = workingSetAfter[i] - workingSetBefore[i];
+                }
+
+                Console.WriteLine("diff[99] = {0}", diff[numIterations-1]);
```

Comment:
Tests should not be writing to the console.

Reply:
Good catch!